src/parse/asp/builtins.go

```diff
@@ -155,6 +159,57 @@ func buildRule(s *scope, args []pyObject) pyObject {
 	return pyString(":" + target.Label.Name)
 }
 
+// doExec fork/exec's a command and returns the output as a string.  exec
+// accepts either a string or a list of commands and arguments.
+func doExec(s *scope, args []pyObject) pyObject {
+	cmdIn := args[0]
+	wantStdout := args[1].IsTruthy()
+	wantStderr := args[2].IsTruthy()
+
+	var cmdArgs []string
+	if isType(cmdIn, "str") {
+		cmdArgs = strings.Fields(string(cmdIn.(pyString)))
+	} else if isType(cmdIn, "list") {
+		pl := cmdIn.(pyList)
+		cmdArgs = make([]string, 0, pl.Len())
+		for i := 0; i < pl.Len(); i++ {
+			cmdArgs = append(cmdArgs, pl[i].String())
+		}
+	}
+
+	ctx, cancel := context.WithCancel(context.TODO())
```

Comment:
I guess a potential use for this context could be allowing a timeout arg to this exec so that the user can get a build interrupted if they specify a timeout (to give them better output / errors in case the command is executing forever).

Reply:
Correct.  I wasn't sure how to pull down a timeout and use it here.  I've switched to using a timeout from the state for now.  It's not entirely correct, but I'm not sure where to get the current target from the `scope`.

Reply:
I can add that in a second pass if you'd like.  I found *a* way of pulling in some timeout value, but I'm not sure it's right.